covid19_sfbayarea/data/solano.py

```diff
@@ -158,129 +142,105 @@ def get_notes() -> str:
-    response = requests.get(age_group_url, params=param_list)
+
+     # filter for any days on which a total age group cases was reported
+    param_list = {'where': "Age_group='Total_AG'",'outFields': 'Date_reported', 'orderByFields':'Date_reported DESC', 'resultRecordCount': '1', 'f': 'json'}
+    response = requests.get(data2_url, params=param_list)
     response.raise_for_status()
     parsed = response.json()
-    # surface data from nested attributes dict
-    entries = [attr["attributes"] for attr in parsed['features']]
+    latest_day_timestamp = parsed['features'][0]['attributes']['Date_reported']
+    # translate timestamp  to a date string in format 'mm-dd-yyyy' to use in the query string
+    latest_day = datetime.fromtimestamp(latest_day_timestamp/1000, tz=timezone.utc).strftime('%m-%d-%Y')
 
-    if len(entries) != 4: # check that we have 4 entries, one for each expected age group
-        raise FormatError(
-            f"The source data structure has changed. Query did not return four age groups. Results: {entries}")
+    # get all positive values for race/ethnicity total cases on the latest day
+    param2_list = {'where': f"AG_Total_cases>0 AND Date_reported = '{latest_day}' ",'outFields': 'Age_group, AG_Total_cases, AG_deaths', 'f': 'json'}
+    response2 = requests.get(data2_url, params=param2_list)
+    response2.raise_for_status()
+    parsed2 = response2.json()
 
-    # Dict of source_label: target_label for re-keying.
-    AGE_KEYS = {"0-17 yrs": "0_to_17",
-                "18-49 yrs": "18_to_49", "50-64 yrs": "50_to_64", "65+ yrs": "65_and_older" }
-    age_table_cases = []
-    age_table_deaths = []
+    age_group_cases = { entry['attributes']['Age_group']: entry['attributes']['AG_Total_cases']  for entry in parsed2['features'] }
+    # A complete table will have 5 datapoints, as of 9/18/20. If there are any more or less, raise an error.
+    if len(age_group_cases) != 5:
+        raise FormatError( f'Race_eth query did not return 5 groups. Results: {age_group_cases}')
 
-    # parse output
-    for entry in entries:
-        age_key = AGE_KEYS[entry["Age_Group"]]
-        age_group_cases = entry["All_cases_Number"] or 0 # explicitly set 0 for null values
-        age_group_deaths = entry["Died_Number"] or 0
-        age_table_cases.append(
-            {"group": age_key, "raw_count": age_group_cases})
-        age_table_deaths.append(
-            {"group": age_key, "raw_count": age_group_deaths})
+    age_group_deaths = { entry['attributes']['Age_group']: entry['attributes']['AG_deaths']  for entry in parsed2['features'] }
+    # save to the out dict
+    out["case_totals"]["age_group"] = age_group_cases
+    out["death_totals"]["age_group"] = age_group_deaths
 
-    out["case_totals"]["age_group"] = age_table_cases
-    out["death_totals"]["age_group"] = age_table_deaths
+    out["case_totals"]["age_group"] = age_group_cases
+    out["death_totals"]["age_group"] = age_group_deaths
```

Comment:
Looks like we are just doing the same thing twice here.

---

Also, this is outputting:

```js
"age_group": {
  "0_17_yrs": 661,
  "18_49_yrs": 3626,
  "50_64_yrs": 1160,
  "65_pl_yrs": 606,
  "Total_AG": 6054
},
```

But the format should be:

```js
"age_group": [
  {"group": "0_17_yrs", "count": 661},
  {"group": "18_49_yrs", "count": 3626},
  {"group": "50_64_yrs", "count": 1160},
  {"group": "65_pl_yrs", "count": 606},
  {"group": "Total_AG", "count": 6054}
]
```